Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -1,4 +1,7 @@
 #if compiler(>=5.5) && canImport(_Concurrency) && os(Linux)
+// we shouldn't have to check the operating system here, but there is a bug currently where on older macOS versions
```

Comment:
Once we do have support for macOS 12+ in CI, how will we handle these availability checks? Will we just drop the affected versions from the matrix? Also, do we want to add a TODO here for the macOS 12 ticket?

Reply:
I think the bug fix would come in a Swift 5.5 patch release. 

if it is out by whenever that is, we could just bump our Swift 5.5 version in CI on all macOS versions and these should get properly skipped.

if it is not fixed, then we could use a filter with `swift test` so that we would skip all of these on old macOS versions. this is [what Vapor does](https://github.com/vapor/vapor/blob/6a5a3b5244d39e2614382c77ddf62e63b712ad06/.github/workflows/test.yml#L75). the crashes only happens when the test cases get wrongly executed, not when compiling or running other tests, so the bug doesn't block everything else from executing. the only annoyance there is that for local testing on macOS 11, all of you who have not upgraded would have to remember to always provide a filter to `swift test` (I guess we could add something to the makefile for this to make it slightly easier). 

and yeah good idea on the TODO, added and also added mention of the issue you point out here / workarounds.